packages/genesys-spark-components/src/components/beta/gux-avatar-group/gux-avatar-groups.tsx

```diff
@@ -0,0 +1,283 @@
+import {
+  Component,
+  h,
+  JSX,
+  Element,
+  State,
+  readTask,
+  Prop,
+  writeTask
+} from '@stencil/core';
+import { afterNextRenderTimeout } from '@utils/dom/after-next-render';
+import { trackComponent } from '@utils/tracking/usage';
+import { GuxAvatarAccent } from './../gux-avatar/gux-avatar.types';
+import { randomHTMLId } from '@utils/dom/random-html-id';
+
+/**
+ * @slot - Some gux-avatars
+ */
+@Component({
+  styleUrl: 'gux-avatar-group.scss',
+  tag: 'gux-avatar-group-beta',
+  shadow: true
+})
+export class GuxAvatarGroup {
+  @Element()
+  root: HTMLElement;
+
+  private resizeObserver?: ResizeObserver;
+  private listItemClickListeners: Map<HTMLElement, (event: Event) => void> =
+    new Map();
+
+  @Prop()
+  avatarLimit: number = 7;
+
+  @State()
+  avatarList: HTMLGuxAvatarFocusableBetaElement[];
+
+  @State()
+  private overflowCount: number = 0;
+
+  @State()
+  private maxAvatarsToDisplay: number = 0;
+
+  get countAvatars() {
+    return this.avatarList.length ?? 0;
+  }
+
+  async componentWillLoad(): Promise<void> {
+    this.avatarList = Array.from(
+      this.root.children
```

Comment:
Should there be some sort of validation to assure that these are `gux-avatar-focusable-beta` elements or log a warning if they are not ?